addons/test_mail_full/tests/test_rating.py

```diff
@@ -208,11 +210,78 @@ def test_open_rating_route(self):
         for record_rating, is_rating_mixin_test in ((self.record_rating_thread, False),
                                                     (self.record_rating, True)):
             with self.subTest('With rating mixin' if is_rating_mixin_test else 'Without rating mixin'):
+                """
+                16.0 + expected behavior
+                1) Clicking on the smiley image triggers the /rate/<string:token>/<int:rate>
+                route should not update the rating of the record but simply redirect
+                to the feedback form
+                2) Customer interacts with webpage and submits FORM. Triggers /rate/<string:token>/submit_feedback
+                route. Should update the rating of the record with the data in the POST request
+                """
+                self.authenticate(None, None)  # set up session for public user
                 access_token = record_rating._rating_get_access_token()
-                self.url_open(f"/rate/{access_token}/5")
 
+                # First round of clicking the URL and then submitting FORM data
+                response_click_one = self.url_open(f"/rate/{access_token}/5")
+                response_click_one.raise_for_status()
+
+                # there should be a form to post to validate the feedback and avoid one-click anyway
+                forms = lxml.html.fromstring(response_click_one.content).xpath('//form')
+                matching_rate_form = next((form for form in forms if form.get("action", "").startswith("/rate")), None)
+                self.assertEqual(matching_rate_form.get('method'), 'post')
+                self.assertEqual(matching_rate_form.get('action', ''), f'/rate/{access_token}/submit_feedback')
```

Comment:
@tde-banana-odoo had to change the logic here to account for the possibility of multiple form elements

Runbot was failing  for the "All at_install "build. As it happens in 17.0, when installing all modules, you will have form elements for the website search.

Example if you parse all the 'action' slots in the response HTML:
['/website/search', '/website/search', '/rate/c5ab5f47b4514736b4a5bf91866d8526/submit_feedback']

So to make it more specific, I tried to match the first element that has an action stating with "/rate"